source/nodejs/adaptivecards-fabric/src/components/inputs/InputChoiceSetFabric.tsx

```diff
@@ -20,9 +20,8 @@ export class InputChoiceSetFabric extends Shared.ReactInputElement {
     private selectedValues: string[] = [];
 
     public parse = (json: any, errors?: AC.IValidationError[]) => {
-        this.id = json.id ? AC.getStringValue(json.id) : null;
-        this.valueInternal = json.value;
-        this.defaultValue = this.valueInternal;
+        super.parse(json, errors);
+        this.value = AC.getStringValue(json.value, "");
         this.selectedValues = this.defaultValueToArray(this.value);
         this.choices = json.choices;
```

Comment:
Ideally you'd use the Choice class that's already part of the library. The built-in ChoiceSet input parses choices like this, which I suggest you copy. Again that is because Choice extends SerializableObject which handles (de)serialization in a specific way that works well with the designer:

```typescript
        this.choices = [];

        if (Array.isArray(json["choices"])) {
            for (let jsonChoice of json["choices"]) {
                let choice = new Choice();
                choice.parse(jsonChoice);

                this.choices.push(choice);
            }
        }
```